backend/service/chaos/experimentation/experimentstore/experiment_store.go

```diff
@@ -58,61 +50,66 @@ func New(cfg *any.Any, _ *zap.Logger, _ tally.Scope) (service.Service, error) {
 }
 
 func (fs *experimentStore) CreateExperiments(ctx context.Context, experiments []*experimentation.Experiment) error {
-	psql := sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
-	builder := psql.Insert(experimentsTableName).Columns(experimentColumns...)
+	// This API call will eventually be broken into 2 separate calls:
+	// 1) creating the config
+	// 2) starting a new experiment with the config
+
+	// All experiments are created in a single transaction
+	tx, err := fs.db.Begin()
+	if err != nil {
+		return err
+	}
 
 	for _, experiment := range experiments {
-		marshaler := jsonpb.Marshaler{}
-		buf := &bytes.Buffer{}
-		err := marshaler.Marshal(buf, experiment.GetConfig())
+		// Step 1) create the config
+		configID := id.NewID()
+
+		configJson, err := marshalConfig(experiment)
 		if err != nil {
 			return err
 		}
-		s := buf.String()
-		builder = builder.Values(
-			id.NewID(),
-			s,
-		)
-	}
 
-	sql, args, err := builder.ToSql()
-	if err != nil {
-		return err
+		configSql := `INSERT INTO experiment_config (id, details) VALUES ($1, $2)`
+		_, err = fs.db.ExecContext(ctx, configSql, configID, configJson)
+		if err != nil {
+			return err
+		}
+
+		// Step 2) start a new experiment with the config
+		runSql := `
+			INSERT INTO experiment_run (
+                id, 
+		        experiment_config_id,
+		        start_time,
+			    end_time,                        
+		        creation_time)
+            VALUES ($1, $2, NOW(), NULL, NOW())`
+
+		_, err = fs.db.ExecContext(ctx, runSql, id.NewID(), configID)
+		if err != nil {
+			return err
+		}
 	}
 
-	_, err = fs.db.ExecContext(ctx, sql, args...)
+	err = tx.Commit()
 	return err
 }
 
 // DeleteExperiments deletes the specified experiments from the store.
 func (fs *experimentStore) DeleteExperiments(ctx context.Context, ids []uint64) error {
-	psql := sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
-	builder := psql.Delete("").From(experimentsTableName)
-
-	if len(ids) > 0 {
-		builder = builder.Where(sq.Eq{"id": ids})
-	}
-
-	sql, args, err := builder.ToSql()
-	if err != nil {
-		return err
-	}
-
-	_, err = fs.db.ExecContext(ctx, sql, args...)
+	sql := `DELETE FROM experiment_run WHERE id IN $1`
```

Comment:
not a part of this PR: we will need to probably rename this method to something like `StopExperiment` instead and avoid removing the records from SQL.

Reply:
Yeah, I'd like to have a separate PR that's dedicated mostly to API changes.

Reply:
https://jira.lyft.net/browse/RES-1551